Improvement of TestCopier

yummypeng opened the pull request:
Since `Copier.Run()` won't return any error, it would be confusing to
users if an error is returned when `Decode` in `TestCopier()` but it
actually comes from `Copier.Run()`. So it's better to check if an
error occurred during the process of `Copier.Run()` before `Decode`.

Signed-off-by: Yuanhong Peng <pengyuanhong@huawei.com>

AkihiroSuda commented on daemon/logger/copier.go:
```diff
@@ -23,6 +23,8 @@ type Copier struct {
 	closeOnce sync.Once
 	closed    chan struct{}
+	// err is the error occurs during `Copier.Run()`
+	err error
```
seems racy

yummypeng replied:
Thank you for code review. :stuck_out_tongue_winking_eye: 
I changed it to be a map to avoid race. PTAL @AkihiroSuda

AkihiroSuda replied:
we still need a mutex

yummypeng replied:
Sorry for misunderstanding what you meant. I add a mutex now.

Commit:
Improvement of TestCopier

Since `Copier.Run()` won't return any error, it would be confusing to
users if an error is returned when `Decode` in `TestCopier()` but it
actually comes from `Copier.Run()`. So it's better to check if an
error occurred during the process of `Copier.Run()` before `Decode`.

Signed-off-by: Yuanhong Peng <pengyuanhong@huawei.com>

cpuguy83 commented:
I don't think we should take this.
Errors here are not able to be propagated anywhere other than logging them.

As far as the tests go, these errors are not related to the copier itself and out of the control of the package, so it seems a little weird account for them in the tests.

yummypeng commented:
Oh, I see..Close it !
